x-pack/plugins/alerting/server/task_runner/ad_hoc_task_runner.ts

```diff
@@ -573,6 +573,10 @@ export class AdHocTaskRunner {
             : undefined,
       },
     });
+    this.shouldDeleteTask = !this.hasAnyPendingRuns();
+
+    // cleanup function is not called for timed out tasks
+    await this.cleanup();
```

Comment:
I was curious who was calling `cancel()` and confused when VS Code found no references.  I _think_ what's going on is instances of AdHocTaskRunner actually conform to `CancellableTask` based on the naming, and so is used as one of those instances.  Once I added `implements CancellableTask` to the `AdHocTaskRunner` class definition, all references in VSCode turned up the calls in task manager.

So, assuming I'm right, let's add the `implements` bit so we'll get a little better searchability in VS Code (and presumably other editors).

Reply:
Updated in https://github.com/elastic/kibana/pull/187496/commits/55158549041b1cd26f0d6eae702260402b81ccbf